### ISSUE TITLE
Tests (randomly) failing

### ISSUE THREAD

**Issue by LDAP** [SMTorg/smt#272]
- GitHub CI fails randomly 
- I had locally never a run with all tests passed, [pytest.log](https://github.com/SMTorg/smt/files/5798046/pytest.log), commit: 9f6610ef31b37cc2a1bfaec7fe1a0f24610f550d
I am using Windows 10, Python 3.8.3.

**Comment by Paul-Saves** [comment]
Have you tried it after #269 and #270  ? (merge today)

**Comment by relf** [comment]
I reproduce this kind of errors with scikit-learn 0.24. It works fine after downgrading to sklearn 0.23.2 (on Windows 7, python 3.7). Could you try?

**Comment by LDAP** [comment]
I can confirm: All tests passing with scikit-learn 0.23.2.
Is this an upstream bug in scikit-learn or fixes a 0.24 a bug in 0.23.2 which led to false results?

In any case I would prefer to enforce the dependency version ether via setup.py install_requires or at least to add a requirements.txt to the repo, that the developers who mostly run the tests do have a consistent development environment (This file could also define the dependencies to run the setup (numpy and Cython)).

**Comment by relf** [comment]
> Is this an upstream bug in scikit-learn or fixes a 0.24 a bug in 0.23.2 which led to false results?

We just discover the problem... To be investigated...

> In any case I would prefer to enforce the dependency version ether via setup.py install_requires or at least to add a requirements.txt to the repo, that the developers who mostly run the tests do have a consistent development environment (This file could also define the dependencies to run the setup (numpy and Cython)).

I am a bit reluctant to fix a required version. If we had here we would have discovered the problem maybe much later and scikit-learn 0.24 seems to be ok for surrogate models without PLS, so maybe a requirements.txt... and you're right <code>setup.py</code> has to be updated.

**Comment by relf** [comment]
superseded by #274
